src/groupByKey.ts

```diff
@@ -2,6 +2,13 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], searchedKey: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+  const searchedKeys = new Set(items.map((value) => value[searchedKey]));
+
+  searchedKeys.forEach((key) => {
+    result[key] = items.filter((item) => item[searchedKey] === key);
```

Comment:
The usage of the filter method inside a forEach loop can lead to performance issues because it iterates over the entire array for each key. Consider using a single loop for efficiency.